load/gradual_orchestrator_test.go

```
@@ -192,47 +162,40 @@ func TestGradualLoadOrchestratorExecution(t *testing.T) {
 	}
 }
 
-type mockTxGenerator struct {
-	generateTxF func() (ids.ID, error)
-}
-
-func (m *mockTxGenerator) GenerateTx(context.Context) (ids.ID, error) {
-	return m.generateTxF()
-}
-
 type mockIssuer struct {
+	generateTxF   func() (ids.ID, error)
 	currTxsIssued uint64
 	maxTxs        uint64
 	tracker       Tracker[ids.ID]
 	issueTxErr    error
 }
 
-func newMockIssuer(tracker Tracker[ids.ID], maxTxs uint64) *mockIssuer {
-	return &mockIssuer{
-		tracker: tracker,
-		maxTxs:  maxTxs,
+// GenerateAndIssueTx immediately generates, issues and confirms a tx.
+// To simulate TPS, the number of txs IssueTx can issue/confirm is capped by maxTxs
+func (m *mockIssuer) GenerateAndIssueTx(_ context.Context) (ids.ID, error) {
+	id, err := m.generateTxF()
+	if err != nil {
+		return id, err
 	}
-}
 
-// IssueTx immediately issues and confirms a tx
-// To simulate TPS, the number of txs IssueTx can issue/confirm is capped by maxTxs
-func (m *mockIssuer) IssueTx(_ context.Context, id ids.ID) error {
 	if m.issueTxErr != nil {
-		return m.issueTxErr
+		return ids.ID{}, m.issueTxErr
 	}
 
 	if m.currTxsIssued >= m.maxTxs {
-		return nil
+		return ids.ID{}, nil
```

Should this return a non-nil error as opposed to returning an empty ID? Unless an empty ID is specifically checked (which it won't be since this is a generic type from the perspective of the upstream component), then this will look like successful generation of a new tx.

If I'm misunderstanding, unless it's for a very simple reason I'm missing, would it make sense to add a comment in the code explaining as it seems non-obvious to me?

---

This should return a nil error. The point of `maxTxs` is to "simulate" the TPS of the network and so to simulate the case of there being an upper bound on TPS, we mock a successful TX issuance to the orchestrator. Errors are mocked in the `if`- statement above.

---

Hmm seems this is not a change from before, but could produce confusing behavior to the orchestrator